src/packages/async/async.cc

```diff
@@ -2,18 +2,32 @@
 
 #include "packages/async/async.h"
 
+#include <chrono>
+#include <deque>
+#include <mutex>
+#include <thread>
+
+#if HAVE_DIRENT_H
 #include <dirent.h>
+#define NAMLEN(dirent) strlen((dirent)->d_name)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126).  ](https://app.codacy.com/gh/fluffos/fluffos/pullRequest?prid=5214413)